src/redux/books/bookSlice.js

```diff
@@ -0,0 +1,34 @@
+const ADD_BOOK = 'bookstore/books/addBook';
+const REMOVE_BOOK = 'bookstore/books/removeBook';
+
+const initialState = {
+  books: [],
+};
+
+// Add Reducers
+export default function booksReducer(state = initialState, action) {
+  switch (action.type) {
+    case ADD_BOOK:
+      return {
+        ...state.books,
+        books: action.payload,
+      };
```

Comment:
- [ ] Dear @edubew, I see you have rearranged this case, But please notice by setting the `books: action. payload,` You are replacing all your previous books with the new book. Instead, we need to return an  array of books with the new book
> Something like:

````javascript
return {...state, books: [...state.books, action.payload]}

// or 

state.books = [...state.books, action.payload]]
````